### ISSUE TITLE
na_ontap_command return_dict throws an error

### ISSUE THREAD

**Issue by nikolaykanov** [ansible-collections/netapp#39]
##### SUMMARY

na_ontap_command return_dict always gives me an error:
parse_xml_to_dict\nValueError: invalid literal for int() with base 10: 'u1'\n"

##### ISSUE TYPE
- Bug Report

##### ANSIBLE VERSION

```paste below
ansible 2.9.1
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Aug 29 2016, 10:12:21) [GCC 4.8.5 20150623 (Red Hat 4.8.5-4)]
```

##### CONFIGURATION

```paste below
HOST_KEY_CHECKING(/etc/ansible/ansible.cfg) = False
```

##### OS / ENVIRONMENT

NAME="Red Hat Enterprise Linux Server"
VERSION="7.2 (Maipo)"

target: NetApp Release 9.3P7: Wed Jul 25 10:11:10 UTC 2018

##### STEPS TO REPRODUCE

```yaml
- hosts: localhost
  gather_facts: False
  vars:
    hostname: xxxxxxx
    username: xxxxxxx
    password: xxxxxxx

  tasks:
  - name: run ontap cli command
    na_ontap_command:
      hostname: "{{ hostname }}"
      username: "{{ username }}"
      password: "{{ password }}"
      command: "version"
      validate_certs: no
      https: true
      return_dict: yes
    register: command_out

  - name: print output
    debug:
      msg: "{{ command_out.msg }}"
```

##### EXPECTED RESULTS

this is what happens when I do it without return_dict:

PLAY [localhost] ****************************************************************************************************************************************************************************

TASK [run ontap cli command] ****************************************************************************************************************************************************************
changed: [localhost]

TASK [print output] *************************************************************************************************************************************************************************
ok: [localhost] => {
    "msg": "<results xmlns=\"http://www.netapp.com/filer/admin\" status=\"passed\"><cli-output>NetApp Release 9.3P14: Wed Jul 10 14:38:46 UTC 2019\n\n</cli-output><cli-result-value>1</cli-result-value></results>"

I expect the same output in dictionary form

##### ACTUAL RESULTS

```paste below
PLAY [localhost] ****************************************************************************************************************************************************************************

TASK [run ontap cli command] ****************************************************************************************************************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ValueError: invalid literal for int() with base 10: 'u1'
fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"/root/.ansible/tmp/ansible-tmp-1589532394.97-154348621604878/AnsiballZ_na_ontap_command.py\", line 102, in <module>\n    _ansiballz_main()\n  File \"/root/.ansible/tmp/ansible-tmp-1589532394.97-154348621604878/AnsiballZ_na_ontap_command.py\", line 94, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/root/.ansible/tmp/ansible-tmp-1589532394.97-154348621604878/AnsiballZ_na_ontap_command.py\", line 40, in invoke_module\n    runpy.run_module(mod_name='ansible.modules.storage.netapp.na_ontap_command', init_globals=None, run_name='__main__', alter_sys=True)\n  File \"/usr/lib64/python2.7/runpy.py\", line 176, in run_module\n    fname, loader, pkg_name)\n  File \"/usr/lib64/python2.7/runpy.py\", line 82, in _run_module_code\n    mod_name, mod_fname, mod_loader, pkg_name)\n  File \"/usr/lib64/python2.7/runpy.py\", line 72, in _run_code\n    exec code in run_globals\n  File \"/tmp/ansible_na_ontap_command_payload_xYTFFO/ansible_na_ontap_command_payload.zip/ansible/modules/storage/netapp/na_ontap_command.py\", line 228, in <module>\n  File \"/tmp/ansible_na_ontap_command_payload_xYTFFO/ansible_na_ontap_command_payload.zip/ansible/modules/storage/netapp/na_ontap_command.py\", line 224, in main\n  File \"/tmp/ansible_na_ontap_command_payload_xYTFFO/ansible_na_ontap_command_payload.zip/ansible/modules/storage/netapp/na_ontap_command.py\", line 149, in apply\n  File \"/tmp/ansible_na_ontap_command_payload_xYTFFO/ansible_na_ontap_command_payload.zip/ansible/modules/storage/netapp/na_ontap_command.py\", line 135, in run_command\n  File \"/tmp/ansible_na_ontap_command_payload_xYTFFO/ansible_na_ontap_command_payload.zip/ansible/modules/storage/netapp/na_ontap_command.py\", line 187, in parse_xml_to_dict\nValueError: invalid literal for int() with base 10: 'u1'\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}

PLAY RECAP **********************************************************************************************************************************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0

```

**Comment by zeten30** [comment]
Hi @nikolaykanov,

can you share your locale settings. This is most probably related to UTF-8/Unicode stuff in Python. u'<something>' is marking unicode string.  

I'm trying to reproduce, but works without issues for me:

```
  tasks:
    - name: na_ontap_command - standard XML output
      na_ontap_command:
        command: ["version"]
        # command:      ['version']
        https: true
        validate_certs: false
        hostname: "{{ inventory_hostname }}"
        username: "{{ admin_username }}"
        password: "{{ admin_password }}"
      register: cmd_out_xml

    - name: Results of na_ontap_command - standard XML output
      debug:
        var: cmd_out_xml
```

```
ansible-playbook test_command_module.yml --limit nlab-rdu2-c01.mgmt.soclab.redhat.com

PLAY [all] ********************************************************************************************************************************

TASK [na_ontap_command - standard XML output] *********************************************************************************************
changed: [nlab-rdu2-c01.mgmt.soclab.redhat.com]

TASK [Results of na_ontap_command - standard XML output] **********************************************************************************
ok: [nlab-rdu2-c01.mgmt.soclab.redhat.com] => 
  cmd_out_xml:
    changed: true
    failed: false
    msg: |-
      <results xmlns="http://www.netapp.com/filer/admin" status="passed"><cli-output>NetApp Release 9.6P4: Thu Nov 07 01:50:36 UTC 2019
  
      </cli-output><cli-result-value>1</cli-result-value></results>

PLAY RECAP ********************************************************************************************************************************
nlab-rdu2-c01.mgmt.soclab.redhat.com : ok=2    changed=1    unreachable=0    failed=0    skipped=1    rescued=0    ignored=0 
```

**Comment by zeten30** [comment]
Bad examples, sorry. Here's the correct output:

```
  tasks:
    - name: na_ontap_command - standard XML output
      na_ontap_command:
        command: ["version"]
        # command:      ['version']
        https: true
        validate_certs: false
        hostname: "{{ inventory_hostname }}"
        username: "{{ admin_username }}"
        password: "{{ admin_password }}"
        return_dict: true
      register: cmd_out_xml

    - name: Results of na_ontap_command - standard XML output
      debug:
        var: cmd_out_xml
```

```
ansible-playbook test_command_module.yml --limit nlab-rdu2-c01.mgmt.soclab.redhat.com

PLAY [all] ********************************************************************************************************************************

TASK [na_ontap_command - standard XML output] *********************************************************************************************
changed: [nlab-rdu2-c01.mgmt.soclab.redhat.com]

TASK [Results of na_ontap_command - standard XML output] **********************************************************************************
ok: [nlab-rdu2-c01.mgmt.soclab.redhat.com] => 
  cmd_out_xml:
    changed: true
    failed: false
    msg:
      invoked_command: version
      result_value: 1
      status: passed
      stdout: |-
        NetApp Release 9.6P4: Thu Nov 07 01:50:36 UTC 2019
      stdout_lines:
      - 'NetApp Release 9.6P4: Thu Nov 07 01:50:36 UTC 2019'
      xml_dict:
        active_element: ''
        cli-output:
          attrs: {}
          data: |-
            NetApp Release 9.6P4: Thu Nov 07 01:50:36 UTC 2019
        cli-result-value:
          attrs: {}
          data: '''1'''
        last_element: results
        results:
          attrs:
            status: passed
            xmlns: http://www.netapp.com/filer/admin
          data: ''
```

```
$ ansible --version
ansible 2.9.9
  config file = /home/brq/mzink/Projects/RH/NetApp/ansible-cdot-mgmt/ansible.cfg
  configured module search path = ['/home/brq/mzink/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.8/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.8.2 (default, Feb 28 2020, 00:00:00) [GCC 10.0.1 20200216 (Red Hat 10.0.1-0.8)]
```

```
$ locale
LANG=en_US.UTF-8
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC=en_GB.UTF-8
LC_TIME=en_GB.UTF-8
LC_COLLATE="en_US.UTF-8"
LC_MONETARY=en_GB.UTF-8
LC_MESSAGES="en_US.UTF-8"
LC_PAPER=en_GB.UTF-8
LC_NAME="en_US.UTF-8"
LC_ADDRESS="en_US.UTF-8"
LC_TELEPHONE="en_US.UTF-8"
LC_MEASUREMENT=en_GB.UTF-8
LC_IDENTIFICATION="en_US.UTF-8"

```

**Comment by zeten30** [comment]
I'd suggest to use latest module version from collections & and python3.

https://developers.redhat.com/blog/2018/08/13/install-python3-rhel/

**Comment by nikolaykanov** [comment]
Hi Milan,

Thank you for looking into this so quickly!
I'll do some updates on Monday and try again. I don't have the option of switching to pyhton 3, but I will make sure python 2 is up to date and post the new results here.

In the meantime, here's the locale settings for the sake of completeness:
```
# locale
LANG=en_US.UTF-8
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_PAPER="en_US.UTF-8"
LC_NAME="en_US.UTF-8"
LC_ADDRESS="en_US.UTF-8"
LC_TELEPHONE="en_US.UTF-8"
LC_MEASUREMENT="en_US.UTF-8"
LC_IDENTIFICATION="en_US.UTF-8"
```

**Comment by carchi8py** [comment]
@nikolaykanov looks like your using the Ansible 2.9 version of the modules and not the collection version.

This was fixed in the collection version 20.2.0 
https://github.com/ansible-collections/netapp/blob/master/ansible_collections/netapp/ontap/README.md#bug-fixes-5

**Comment by nikolaykanov** [comment]
Using collection 20.5.0, it works fine. Thank you for your support!
